packages/gatsby/src/redux/nodes.ts

```diff
@@ -761,7 +776,9 @@ export const getNodesFromCacheByValue = (
       if (cache) cache.forEach(node => set.delete(node))
     }
 
-    return set
+    // TODO: there's probably a more efficient algorithm to do set
+    //       subtraction in such a way that we dont have to resort here
+    return [...set].sort((A, B) => (A.id < B.id ? -1 : A.id === B.id ? 0 : 1))
```

Comment:
Looks like `nin` and `ne` have a similar logic for diffing lists of nodes. Maybe we can extract it to a separate utility like `diffNodes` similar to `intersectNodes`?

Reply:
Good point :)